virttest/libvirt_vm.py

```diff
@@ -281,6 +281,97 @@ def is_autostart(self):
         except process.CmdError:
             return False
 
+    def set_state(self, state, shutoff_method="destroy", debug=False):
+        """
+        Set VM state to "running", "os-booted, "paused", "pmsuspended" or "shutoff"
+
+        param state, the target state of VM
+        param shutoff_method: The method to convert VM state to shutoff, it
+        could be "destroy" or "shutdown"
+        param debug: Enable debug log in logging
+        """
+        target_states = ["running", "os-booted", "paused", "pmsuspended", "shutoff"]
+        abnormal_states = ["no state", "idle", "crashed"]
+        cur_state = self.state()
+        if self.wait_for_login():
+            cur_state = "os-booted"
+
+        if cur_state == "in shutdown":
+            time.sleep(30)
+            if self.state() == "in shutdown":
+                logging.error("VM %s still keep in 'in shutdown' state after 30s waiting",
+                              self.name)
+
+        if cur_state in abnormal_states:
+            logging.error("The VM %s's state %s is abnormal", self.name, cur_state)
+            return
+
+        if states not in target_states:
+            logging.error("Unable to set VM %s's state to %s", self.name, state)
```

Comment:
How about we add more info here, to let users know which states are supported?
Something like `'state xxx is not one of %s' % target_states`

Reply:
Isn't it rather `"Setting VM %s to state %s is not supported."`?

Reply:
Thanks for your suggestions. I will combine these two together, then the error message will be like:
"Unable to set state %s from %s: the target state is not the one of %s" %(cur_state, state, target_states)